### PR TITLE
Allow annotated field with JsonKey.unknownEnumValue to be of type List

### PR DESCRIPTION
**Pull request by jpelgrim**
This fixes https://github.com/google/json_serializable.dart/issues/585 and allows us to annotate lists with a default enum value when we encounter an unknown value.

Example:
```dart
  @JsonKey(unknownEnumValue: YourEnum.yourUnkownValue)
  final List<YourEnum> lastOrder;
```
This is currently not possible.

### CODE REVIEW

**Comment by jpelgrim** [comment]
@kevmoo can you please review? We need this PR to be able to generate `null` values for unknown enums in a list.

**Comment by kevmoo** [comment]
@jpelgrim – we need to bump the version and add an entry to the changelog

Also, need to add tests.

Also, I want to understand what the failure modes are here. Is this for filling in lists of enums?

**Comment by jpelgrim** [comment]
Yes, this is for filling in lists of enums? #585 asks for `unknownEnumValue` support for collections, but I just implemented `List` support in this PR. That at least is an improvement and solves the issue we have in our project.

I'm happy to bump the version, add an entry to the changelog and add tests. I might need some help with the latter, so if you have any tips or pointers on the testing part I'd appreciate it.

**Comment by kevmoo** [comment]
Thanks! I'm going to pull this down and play with it. I worry there are some edge cases that need to be covered.

File: json_serializable/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 - Support properties where the getter is defined in a class with a corresponding
   setter in a super type.
+- `JsonKey.unknownEnumValue`
```
Comment (kevmoo) [inline json_serializable/CHANGELOG.md:5]: Please move this to a new entry under `3.4.1-dev` – also update the pubspec to have that version `3.4.1-dev`

File: json_serializable/lib/src/utils.dart [hunk 2]
```diff
@@ -118,6 +118,8 @@ JsonSerializable mergeConfig(JsonSerializable config, ConstantReader reader) {
 bool isEnum(DartType targetType) =>
     targetType is InterfaceType && targetType.element.isEnum;
 
+bool isList(DartType targetType) => targetType.isDartCoreList;
```
Comment (kevmoo) [inline json_serializable/lib/src/utils.dart:121]: Delete this new helper. Just put this `isDartCoreList` call inline above.

File: json_serializable/CHANGELOG.md (hunk 1 above)
Comment (jpelgrim) [reply json_serializable/CHANGELOG.md:5]: Or `3.4.2-dev` since you're already getting ready for the `3.4.1` release on `master`?

Comment (jpelgrim) [reply json_serializable/CHANGELOG.md]: Set it to `3.4.1-dev` as requested. Please let me know if this was a typo and you meant `3.4.2-dev` instead.

File: json_serializable/lib/src/utils.dart (hunk 2 above)
Comment (jpelgrim) [reply json_serializable/lib/src/utils.dart]: Done!

File: json_serializable/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 3.4.1-dev
```
Comment (kevmoo) [inline json_serializable/CHANGELOG.md:1]:
```suggestion
## 3.4.2-dev
```

Sorry!

File: json_serializable/pubspec.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: json_serializable
-version: 3.4.1
+version: 3.4.1-dev
```
Comment (kevmoo) [inline json_serializable/pubspec.yaml:2]:
```suggestion
version: 3.4.2-dev
```

**Pull request merged**